File: Python_Codes/findAunt.py
tupleList1=[('parent', 'Hasib', 'Rakib'),('parent', 'Rakib', 'Sohel'),
            ('parent', 'Rakib', 'Rahim'),('parent', 'Rakib', 'Rebeka'),
            ('parent', 'Sohel', 'Salam'),('parent', 'Rashid', 'Hasib'),
            ('parent','Rebeka','Urmi'),('parent','Urmi','Mousumi')]

male=['Hasib','Rakib','Sohel','Rahim','Salam','Rashid']
female=['Rebeka','Urmi','Mousumi']

X=str(input("Aunt:"))
output=''

for i in range(len(tupleList1)):
    if (tupleList1[i][0] == 'parent'):
        if(tupleList1[i][2] == X):
            for j in range(len(tupleList1)):
                if ((tupleList1[j][0] == 'parent') &(tupleList1[i][1] == tupleList1[j][1]) & (i!=j) & (tupleList1[i][2] in female)):
                    for l in range(len(tupleList1)):
                        if(tupleList1[l][0]=='parent'):
                            if(tupleList1[l][1]==tupleList1[j][2]):
                                output+=tupleList1[l][2]+' '

if(len(output)>0):
    print('Niece/Nephews:',output)
else:
    print('Niece/Nephews:','Not Found')

File: Prolog_Codes/findAunt.pl
parent('Hasib' , 'Rakib'). parent('Rakib' , 'Sohel').
parent('Rakib' , 'Rahim'). parent('Rakib','Rebeka').
parent('Sohel','Salam'). parent('Rebeka','Urmi').
parent('Urmi','Mousumi'). parent('Rashid' , 'Hasib').

male('Rakib'). male('Sohel'). male('Rahim'). male('Hasib').
male('Rahim'). male('Salam'). male('Rashid').

female('Rebeka'). female('Urmi'). female('Mousumi').

sister(X,Y):-parent(Z,X), parent(Z,Y), female(X), not(X=Y).

aunt(X,Y):-sister(X,Z), parent(Z,Y).
